modules/dbnd/src/dbnd/_core/task_run/task_run_sync_local.py

```diff
@@ -164,6 +172,18 @@ def _sync_remote_to_local(self, remote_target, local_target):
         if dbnd_meta_cache.expired or not local_target.exists():
             # If TTL is invalid, or local file doesn't exist -> Sync to local
             local_target.mkdir_parent()
+            _, tmp_file_path = mkstemp()
             remote_target.download(
-                local_target.path, overwrite=local_target.config.overwrite_target,
+                tmp_file_path, overwrite=local_target.config.overwrite_target,
             )
+            try:
+                local_target.move_from(tmp_file_path)
+            except FileAlreadyExists as e:
+                logger.warning(
+                    "Moving file from %s to %s failed. File already exist! | Error: %s"
+                    % (tmp_file_path, local_target.path, e)
+                )
+                os.remove(tmp_file_path)
```

Comment:
os.remove must be placed within try-catch

Reply:
are you sure? we only need to remove file if moving file fails on `FileAlreadyExists`, there's no need to remove it after successful move as it's already not there